### PR TITLE
Docs: fix `generate`-related rendering issues

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

See title :)

### CODE REVIEW

File: docs/source/en/generation_strategies.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ more. It also plays a role in a variety of mixed-modality applications that have
 and vision-to-text. Some of the models that can generate text include
 GPT2, XLNet, OpenAI GPT, CTRL, TransformerXL, XLM, Bart, T5, GIT, Whisper.
 
-Check out a few examples that use [`~transformers.generation_utils.GenerationMixin.generate`] method to produce
+Check out a few examples that use [`~generation.GenerationMixin.generate`] method to produce
```
Comment (gante) [inline docs/source/en/generation_strategies.md:24]: The text was not being converted to a link. I've checked in the doc preview that it is now a link:
Screenshot 2024-05-02 at 12 23 01

File: src/transformers/generation/configuration_utils.py [hunk 2]
```diff
@@ -256,25 +255,21 @@ class GenerationConfig(PushToHubMixin):
             `batch_size`. Indicating a list enables different start ids for each element in the batch
             (e.g. multilingual models with different target languages in one batch)
 
-
-        > Generation parameters exclusive to [assistant generation](https://arxiv.org/abs/2211.17192)
+        > Generation parameters exclusive to assistant generation
```
Comment (gante) [inline src/transformers/generation/configuration_utils.py:258]: Links do not work within the doc section header:
Screenshot 2024-05-02 at 12 24 11

File: src/transformers/generation/configuration_utils.py [hunk 3]
```diff
@@ -218,8 +218,7 @@ class GenerationConfig(PushToHubMixin):
             Switch to sequential beam search and sequential topk for contrastive search to reduce peak memory.
             Used with beam search and contrastive search.
 
-
-        > Parameters that define the output variables of `generate`
+        > Parameters that define the output variables of generate
```
Comment (gante) [inline src/transformers/generation/configuration_utils.py:221]: Special formatting with ` does not work here:
Screenshot 2024-05-02 at 12 24 48

**Review by amyeroberts — APPROVED** [review]
Thanks for updating all of these!

**Pull request merged**